java/adapter/jdbc/src/test/java/org/apache/arrow/adapter/jdbc/h2/JdbcToArrowArrayTest.java

```diff
@@ -0,0 +1,372 @@
+      for (int i = 1; i <= rsmd.getColumnCount(); ++i) {
+        assertEquals(Types.ARRAY, rsmd.getColumnType(i));
+      }
+
+      int rowNum = 0;
+
+      while (resultSet.next()) {
+        Array intArray = resultSet.getArray(INT_ARRAY_FIELD_NAME);
+        assertFalse(resultSet.wasNull());
+        try (ResultSet rs = intArray.getResultSet()) {
+          int arrayIndex = 0;
+          while (rs.next()) {
+            assertEquals(intArrays[rowNum][arrayIndex].intValue(), rs.getInt(2));
+            ++arrayIndex;
+          }
+          assertEquals(intArrays[rowNum].length, arrayIndex);
+        }
+
+        Array floatArray = resultSet.getArray(FLOAT_ARRAY_FIELD_NAME);
+        assertFalse(resultSet.wasNull());
+        try (ResultSet rs = floatArray.getResultSet()) {
+          int arrayIndex = 0;
+          while (rs.next()) {
+            assertEquals(floatArrays[rowNum][arrayIndex].floatValue(), rs.getFloat(2), 0.001);
+            ++arrayIndex;
+          }
+          assertEquals(floatArrays[rowNum].length, arrayIndex);
+        }
+
+        Array strArray = resultSet.getArray(STRING_ARRAY_FIELD_NAME);
+        assertFalse(resultSet.wasNull());
+        try (ResultSet rs = strArray.getResultSet()) {
+          int arrayIndex = 0;
+          while (rs.next()) {
+            assertEquals(strArrays[rowNum][arrayIndex], rs.getString(2));
+            ++arrayIndex;
+          }
+          assertEquals(strArrays[rowNum].length, arrayIndex);
+        }
+
+        ++rowNum;
+      }
+
+      assertEquals(rowCount, rowNum);
+    }
+  }
+
+  @Test
+  public void testJdbcToArrow() throws Exception {
+    int rowCount = 4;
+
+    Integer[][] intArrays = generateIntegerArrayField(rowCount);
+    Float[][] floatArrays = generateFloatArrayField(rowCount);
+    String[][] strArrays = generateStringArrayField(rowCount);
+
+    insertRows(rowCount, intArrays, floatArrays, strArrays);
+
+    final JdbcToArrowConfigBuilder builder =
+        new JdbcToArrowConfigBuilder(new RootAllocator(Integer.MAX_VALUE), JdbcToArrowUtils.getUtcCalendar(), false);
+    builder.setArraySubTypeByColumnNameMap(arrayFieldMapping);
+
+    final JdbcToArrowConfig config = builder.build();
+
+    try (ResultSet resultSet = conn.createStatement().executeQuery(QUERY)) {
+      final VectorSchemaRoot vector;
+
+      try {
+        vector = JdbcToArrow.sqlToArrow(resultSet, config);
+      } catch (Exception e) {
+        e.printStackTrace();
+        throw e;
+      }
+
+      assertEquals(rowCount, vector.getRowCount());
+
+      testIntegerList((ListVector) vector.getVector(INT_ARRAY_FIELD_NAME), rowCount, intArrays);
+      testFloatList((ListVector) vector.getVector(FLOAT_ARRAY_FIELD_NAME), rowCount, floatArrays);
+      testStringList((ListVector) vector.getVector(STRING_ARRAY_FIELD_NAME), rowCount, strArrays);
+    }
+  }
+
+  @Test
+  public void testJdbcToArrowWithNulls() throws Exception {
+    int rowCount = 3;
+    Integer[][] intArrays = new Integer[rowCount][];
```

Comment:
here  it might be clearer (or at least more succinct) if you use array initializers https://www.programiz.com/java-programming/multidimensional-array

Reply:
Changed.